pyanaconda/modules/boss/module_manager.py

```diff
@@ -0,0 +1,122 @@
+#
+# module_manager.py: Anaconda DBUS module management
+#
+# Copyright (C) 2017  Red Hat, Inc.  All rights reserved.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+from pyanaconda.dbus import dbus_constants
+
+from pyanaconda.anaconda_loggers import get_module_logger
+log = get_module_logger(__name__)
+
+
+class ModuleManager(object):
+
+    def __init__(self, bus):
+        self._bus = bus
+        self._dbus = self._bus.get('org.freedesktop.DBus', '/org/freedesktop/DBus')
+        self._started_module_services = []
+        self._failed_module_services = []
+        self._addon_module_services = []
+
+    @property
+    def addon_module_services(self):
+        return self._addon_module_services
+
+    @property
+    def expected_module_services(self):
+        expected_modules = []
+        # NOTE: we can just initialize expected_modules with the constant,
+        # bacuse the extend afterwards will then *modify* the constant! :P
+        expected_modules.extend(dbus_constants.ANACONDA_MODULES)
+        expected_modules.extend(self.addon_module_services)
+        return expected_modules
+
+    @property
+    def running_module_services(self):
+        # For our purpose, just this
+        return self._started_module_services
+
+    @property
+    def modules_starting_finished(self):
+        return set(self._started_module_services + self._failed_module_services) == set(self.expected_module_services)
+
+    def _start_service_cb(self, service, returned=None, error=None):
```

Comment:
I would prefer a name `_finish_start_service`.

Reply:
Sure, that makes sense. But keeping the _cb suffix as I think it's good to be able to differentiate callbacks from normal functions on first glance.